### PR TITLE
feat(TDI-42766): Redshift SSO support with Ping Federate

### PR DESCRIPTION
**Pull request by Voievidko**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-42766

**What is the new behavior?**
Login and password fields were added to Redshift components

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Row/HelpClass.javajet [hunk 1]
```diff
@@ -81,7 +81,8 @@
 							jdbcUrl = ElementParameterParser.getValue(ne, "__JDBC_URL__");
 						}
 					}
-					needUserAndPassword = !"SSO".equals(jdbcUrl);
+					//TODO changeme
+					needUserAndPassword = true;
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Row/HelpClass.javajet]: (this is just sample make redshift works for validation our solution on customer side)
seems this should be removed. this maybe break other db components. which support **needUserAndPassword** is **false**
should not change the value here. you can also check the code before we support SSO.

File: main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Input/AbstractDBInputBegin.javajet [hunk 2]
```diff
@@ -73,7 +73,8 @@ imports="
 							jdbcUrl = ElementParameterParser.getValue(ne, "__JDBC_URL__");
 						}
 					}
-					boolean needUserAndPassword = !"SSO".equals(jdbcUrl);
+					//TODO changeme
```
Comment (jzhao-talend) [inline main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Input/AbstractDBInputBegin.javajet]: see https://github.com/Talend/tdi-studio-se/pull/4112/files?w=1#r343504745

File: main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Row/HelpClass.javajet (hunk 1 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Row/HelpClass.javajet]: This was added only for Redshift.
I try to find any other components which can use overload method, but can't.

File: main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Input/AbstractDBInputBegin.javajet (hunk 2 above)
Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Input/AbstractDBInputBegin.javajet]: This was added only for Redshift.
I try to find any other components which can use overload method, but can't.

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/CleanLoginAndPasswordForPostgreSqlForSSO.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import org.talend.core.model.properties.Item;
+import org.talend.cwm.helper.ConnectionHelper;
+import org.talend.designer.core.model.utils.emf.talendfile.NodeType;
+import org.talend.designer.core.model.utils.emf.talendfile.ProcessType;
+
+public class CleanLoginAndPasswordForPostgreSqlForSSO extends AbstractJobMigrationTask {
+
+    @Override
+    public Date getOrder() {
+        GregorianCalendar gc = new GregorianCalendar(2019, 11, 7, 10, 0, 0);
```
Comment (jzhao-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/CleanLoginAndPasswordForPostgreSqlForSSO.java]: **11** should be **10**

Comment (Voievidko) [reply main/plugins/org.talend.repository/src/main/java/org/talend/repository/model/migration/CleanLoginAndPasswordForPostgreSqlForSSO.java]: Oh, months starts from 0.
Changed it :)

File: main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Row/HelpClass.javajet (hunk 1 above)
Comment (jzhao-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Row/HelpClass.javajet]: https://github.com/Talend/tbd-studio-se/blob/master/main/plugins/org.talend.designer.components.bigdata/components/tHiveInput/tHiveInput_begin.javajet#L36
maybe you don't have bd component code on your localside.

**Pull request merged**